Move reset register setup from an arch_init call to within the machine setup
function.

diff --git a/arch/powerpc/platforms/85xx/bcm98548xmc.c b/arch/powerpc/platforms/85xx/bcm98548xmc.c
index 79d4e92..392f231 100644
--- a/arch/powerpc/platforms/85xx/bcm98548xmc.c
+++ b/arch/powerpc/platforms/85xx/bcm98548xmc.c
@@ -114,6 +114,18 @@ static int __init bcm98548xmc_publish_devices(void)
 }
 machine_device_initcall(bcm98548xmc, bcm98548xmc_publish_devices);
 
+static __be32 __iomem *rstcr;
+static __be32 __iomem *rstdr;
+
+static int __init bcm98548xmc_init_rstcr(void)
+{
+	/* map reset control register */
+	printk(KERN_INFO "bcm98548xmc : map reset control register.\n");
+	rstdr = ioremap(get_immrbase() + 0xE0040, 0xff);
+	rstcr = ioremap(get_immrbase() + 0xE0030, 0xff);
+	return 0;
+}
+
 /*
  * Setup the architecture
  */
@@ -151,6 +163,8 @@ static void __init bcm98548xmc_setup_arch(void)
 #endif
 
 	powersave_nap = 0;
+
+	bcm98548xmc_init_rstcr();
 }
 
 static void bcm98548xmc_show_cpuinfo(struct seq_file *m)
@@ -173,18 +187,6 @@ static void bcm98548xmc_show_cpuinfo(struct seq_file *m)
 	seq_printf(m, "Memory\t\t: %d MB\n", memsize / (1024 * 1024));
 }
 
-static __be32 __iomem *rstcr;
-static __be32 __iomem *rstdr;
-static int __init bcm98548xmc_init_rstcr(void)
-{
-	/* map reset control register */
-	printk(KERN_INFO "bcm98548xmc : map reset control register.\n");
-	rstdr = ioremap(get_immrbase() + 0xE0040, 0xff);
-	rstcr = ioremap(get_immrbase() + 0xE0030, 0xff);
-	return 0;
-}
-arch_initcall(bcm98548xmc_init_rstcr);
-
 void bcm98548xmc_restart(char *cmd)
 {
 	local_irq_disable();
